src/lib/templateResource.ts

```diff
@@ -0,0 +1,40 @@
+type ResourceReducer =
+  | ((resource: string, config: Config) => string)
+  | ((resource: string) => string);
+
+// Removes all spaces from string, this allows the other reducers to have more
+// reliable match patterns as they don't need to be concerned about whitespace.
+function removeWhitespace(resource: string): string {
+  return resource.replace(/\s/g, "");
+}
+
+// Replaces the template placeholder <% RANDOM %> with a random string
+function replaceRandom(resource: string): string {
+  const dictionary =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  const dictionaryLength = dictionary.length;
+  let random = "";
+  for (let i = 0; i < 16; i++) {
+    random += dictionary.charAt(Math.floor(Math.random() * dictionaryLength));
+  }
+  return resource.replace("<%RANDOM%>", random);
+}
+
+// Replaces the template placeholder <% TEST_ID %> with the test ID from config
+function replaceTestId(resource: string, config: Config): string {
+  return resource.replace("<%TEST_ID%>", config.test.id);
+}
+
+export default function templateResource(
+  resource: string,
+  config: Config
+): string {
+  // Note: removeWhitespace must come first to ensure the proceeding reducers
+  // are consitent and deterministic.
+  const reducers: ResourceReducer[] = [
+    removeWhitespace,
+    replaceTestId,
+    replaceRandom
```

Comment:
I wonder if it might not be helpful to have a reducer which looks for other `\<\%(\S+)\%\>` patterns where, if the matched inner string is something other than `TEST_ID` or `RANDOM` then it throws an exception and otherwise just emits what it was given?

Reply:
Whilst I think this is a good idea, I think the validation belongs on input/write into the API and is not the responsibility of the client.